### ISSUE TITLE
Cloudflare handler fails when no DNS records are available

### ISSUE THREAD

**Issue by janpieterz** [ebekker/ACMESharp#252]
The Cloudflare handler will hang until CloudFlare blocks the DNS queries because of too many requests.

My guess is that https://github.com/ebekker/ACMESharp/blob/master/ACMESharp/ACMESharp.Providers.CloudFlare/CloudFlareHelper.cs#L96 returns 0 total pages or something alike.

Don't have an empty domain lying around (anymore) to test and fix this, will happily take a look if someone is willing to lend me access to one or if I happen to get one.

**Comment by ebekker** [comment]
Can I ask, how did you discover this issue?  If you don't have an empty domain, I assume you didn't run into it personally?  Was it reported by someone else?

**Comment by janpieterz** [comment]
Sure! We got a new domain and had to get it started, so encountered the issue but then had to quick fix (by adding 1 record) because we had to go live with the domain quick.

**Comment by ebekker** [comment]
I see, so your test case no longer exists.

Basically, I'm going to leave the ticket open for now until the next CF user encounters something similar so we can troubleshoot.

**Comment by janpieterz** [comment]
Agreed! Any time I'll get a new domain under my control that I can empty out I'll jump into this too.
